### ISSUE TITLE
Make edges within grid cells orthogonal

### ISSUE THREAD

**Issue by meanderix** [terrastruct/TALA#84]
![image](https://github.com/user-attachments/assets/df92fe16-0208-40d9-aaae-254df7d9ef1e)
```
grid-rows: 3
grid-columns: 3
grid-gap: 20
1
2
3
4
5
6
7
8
9
1 -> 3 -> 9 -> 7 -> 1
1 -> 9
3 -> 7
```
Looks like the edges internal to a grid cell do not respect orthogonality rules.

**Comment by alixander** [comment]
can you draw your ideal pathing? I think going around the outside will look weird too. I think internal grid edges is a case where most of the time you want just the straight line. If you have many internal grid edges, it quickly ends up like a maze of overlapping routes otherwise.

**Comment by meanderix** [comment]
@alixander 
![image](https://github.com/user-attachments/assets/e1ec7b30-a04a-4098-9ff7-ecd3b0e87626)
Here's another example where I want to achieve a specific grid layout (4x2). Here it would have been super useful with orthogonal edges. Perhaps this could be set optionally through a directive?

(The previous example was a bit hypothetical and I understand that there are also use cases for non-orthogonal edges.)

**Comment by alixander** [comment]
@meanderix Thank you for the example. I suppose grid was cleaner than if you had run this without grid?

Because I feel like the grid is just a suboptimal placement for this use case. This is the best route I can come up with and it doesn't look great.

<img width="1038" alt="Screenshot 2024-08-29 at 11 13 00 AM" src="https://github.com/user-attachments/assets/c9af88b1-f3cf-4a88-8d9b-94ecaa48a516">

I suppose you could increase the gap size so that the routes go in between, but there's still no way to avoid overlap.

**Comment by meanderix** [comment]
![image](https://github.com/user-attachments/assets/67d908b4-d77f-4841-9cf9-28f4cbae091f)
This is also interesting. If you wrap the upper and lower row in its own box, then you will have mixed results with the edges ...

**Comment by alixander** [comment]
In this case it wasn't able to find a route for `Monitor -> Plan`.

You gave my last comment a thumbs up, does that mean the route I proposed is one you'd be satisfied with?

**Comment by meanderix** [comment]
@alixander Well, I would be happy with orthogonal edges. I guess the optimal path depends. Maybe a "dynamic gap" between rows/columns could also be an option in order to allow a connection in between grid cells. (But I can understand that it adds complexity to the implementation.)

**Comment by alixander** [comment]
Idk, even if there's gap, it still overlaps weirdly. I feel like the construction just doesn't allow for good pathing. I believe this is a case where user design goes further than any edge routing can compensate for. Like this seems to me like a better way to do this (`direction: left` on the bottom container)

<img width="481" alt="Screenshot 2024-08-30 at 2 46 08 PM" src="https://github.com/user-attachments/assets/e7b35944-4d67-494c-90f5-292297dd876c">

**Comment by meanderix** [comment]
@alixander 
![image](https://github.com/user-attachments/assets/9079d76e-8711-4225-be9e-1a6697f8a95c)
Some architect diagram tools, such as Archimate, will have these nifty bends when two edges are crossing. I don't know if that would be useful also for D2. (I'm just thinking out loud about possible ways to still support orthogonal edges...)

**Comment by alixander** [comment]
Right, crossing bridges is a feature request: https://github.com/terrastruct/d2/issues/1932
